test/utils.spec.js

```diff
@@ -262,15 +285,30 @@ describe('utils', function() {
 });
 
 /* eslint-disable no-console */
-function handleConsoleWarns(doWithShimmedConsoleWarn, onWarn) {
-  // Shim `console.warn`
-  var _warn = console.warn;
+function handleWarnings(doWithShimmedConsoleWarn, onWarn) {
+  /* eslint-disable no-console */
+  if (typeof process.emitWarning !== 'function') {
+    // Shim `console.warn`
+    var _warn = console.warn;
+    console.warn = onWarn;
+
+    doWithShimmedConsoleWarn();
 
-  console.warn = onWarn;
+    // Un-shim `console.warn`,
+    console.warn = _warn;
+  } else {
+    /* eslint-disable no-inner-declarations */
+    function onProcessWarn(warning) {
+      onWarn(warning.name + ': ' + warning.message);
+    }
 
-  doWithShimmedConsoleWarn();
+    process.on('warning', onProcessWarn);
 
-  // Un-shim `console.warn`
-  console.warn = _warn;
+    doWithShimmedConsoleWarn();
+
+    process.nextTick(function() {
+      process.removeListener('warning', onProcessWarn);
+    })
+  }
 }
 /* eslint-enable no-console */
```

Comment:
Ah yeah, I think you need one of these `eslint-enable` to re-enable the lint if you don't use the per-line syntax.